src/appsettings/settings.py

```diff
@@ -967,3 +974,41 @@ def check(self):
     def transform(self, value):
         """Transform each item in the list."""
         return tuple(self.inner_setting.transform(item) for item in value)
+
+
+class FileSetting(Setting):
+    """
+    File setting.
+
+    Value of this setting is a pathlib.Path instance.
+    """
+
+    def __init__(self, exists=True, mode="r", *args, **kwargs):
+        """
+        Initialization method.
+
+        Note that attributes ``exists`` and ``mode`` are used in ``FileValidator``. For more information check its
+        documentation.
+
+        Args:
+            exists (bool): File already exists.
+            mode (str): Which mode will be the file opened with.
+            name (str): the name of the setting.
+            default (object): default value given to the setting.
+            required (bool): whether the setting is required or not.
+            prefix (str):
+                the setting's prefix (overrides ``AppSettings.Meta`` prefix).
+            call_default (bool): whether to call the default (if callable).
+            transform_default (bool): whether to transform the default value.
+            validators (list of callables): list of additional validators to use.
+        """
+        super(FileSetting, self).__init__(*args, **kwargs)
```

Comment:
Since we use python 3 only, please use only `super()` and add annotations to the new code.

Reply:
It should be done now.